### DISCUSSION TITLE
Is it possible to disable fancy tracebacks?

### DISCUSSION THREAD

**Discussion by SamStudio8** [Textualize/rich#2412] [Q&A] [answered]
While I generally love seeing the use of Rich bloom into all sorts of different applications, today I'm trying to open an issue on a piece of software that uses Rich to render tracebacks. The traceback is quite verbose (taking up several pages of my terminal scrollback), and uses many drawing characters. I would rather just use the basic Python traceback for my report. The software does not seem to offer an option to prevent rendering tracebacks with Rich.

I have suggested the downstream software consider whether that is an option they wish to provide, but given that Rich supports the setting of TERM=dumb and NO_COLOR to disable some of its other features, it would seem reasonable to expect Rich to be able to override the downstream application in this case as well. These existing options don't affect the use of box drawing characters or the traceback verbosity. Have I missed a way to "turn off" this functionality? If not, does the community think this warrants a feature request, or am I just being picky?

**Comment by willmcgugan** [comment, marked as answer]
I'm wary of adding new env vars in to Rich. The reason is that it probably isn't the responsibility of Rich to introduce new inputs in to an application. Rich would essentially be foisting new feature on to the app.

There are also different contexts in which you can capture and print an exception in Rich. They don't necessarily get printed to stdout. An env var might not be granular enough for some use cases.

There's also the issue that if you can switch between rich and standard exceptions, you might also want to be able to configure the show_locals, or max_string, etc. And you end up with a proliferation of env vars that you are again pushing on the app developer, wether they want them or not.

The `TERM` and `NO_COLOR` env vars are pre-existing standards, which is why we support them.

I am sympathetic, and I may yet change my mind if there is some kind of consensus, but it is probably best left to the app developer to implement such a switch.

**Reply by SamStudio8** [reply]
I ultimately agree with you that it is not Rich's responsibility to provide this, but it is unfortunate that it is so easy for developers to paste two lines from the documentation and build applications that generate beautiful tracebacks that can turn out a little unwieldy for end users to make reports, given that is one of the objectives of a traceback.

I suppose it will be up to individual projects using the library to decide whether or not they like the reports coming in with the Rich formatting and make a decision to implement switches for themselves, and I should just paste what I get!
